cloud/scope/powervs_cluster.go

```diff
@@ -2467,9 +2584,10 @@ func (s *PowerVSClusterScope) DeleteVPC() (bool, error) {
 
 // DeleteTransitGateway deletes transit gateway.
 func (s *PowerVSClusterScope) DeleteTransitGateway() (bool, error) {
+	skipTGDeletion := false
 	if !s.isResourceCreatedByController(infrav1beta2.ResourceTypeTransitGateway) {
-		s.Info("Skipping transit gateway deletion as resource is not created by controller")
-		return false, nil
+		s.Info("Skipping transit gateway deletion as resource is not created by controller, but will check if connections are created by the controller.")
+		skipTGDeletion = true
 	}
 
```

Comment:
@dharaneesh, while testing, came across a scenario where for any reason if TG reconcilation fails and it doesn't set the `s.IBMPowerVSCluster.Status.TransitGateway`, it will return a nil pointer error here as ID will be nil. Though the value of `s.IBMPowerVSCluster.Status.TransitGateway` is being checked during `isResourceCreatedByController(infrav1beta2.ResourceTypeTransitGateway)`, we don't return after it to allow deletion of TG connections separately.